packages/coding-standard/src/Rules/NoProtectedElementInFinalClassRule.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Analyser\Scope;
+
+/**
+ * @see \Symplify\CodingStandard\Tests\Rules\NoProtectedElementInFinalClassRule\NoProtectedElementInFinalClassRuleTest
+ */
+final class NoProtectedElementInFinalClassRule extends AbstractManyNodeTypeRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Do not use protected element in final class';
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class, ClassMethod::class];
+    }
+
+    /**
+     * @param Property|ClassMethod $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        $parent = $node->getAttribute('parent');
+        if (! $parent instanceof Class_) {
+            return [];
+        }
+
+        if (! $parent->isFinal()) {
+            return [];
+        }
+
+        if (! $node->isProtected()) {
+            return [];
+        }
+
+        if ($parent->extends !== null) {
+            $name = $this->getName($node);
+
+            if ($node instanceof Property && $parent->extends->getProperty($name) === null) {
+                return [];
+            }
+
+            if ($node instanceof ClassMethod && $parent->extends->getMethod($name) === null) {
+                return [];
+            }
```

Comment:
`$parent->extends` will work only with direct parent.

What if we have extra class in between?  

```php
abstract class AbstractCommand extends Command 
{
}

final class SomeCommand extends AbstractCommand
{
    protected function configure()
	{
    }
}
```

Reply:
ok, it seems we can just skip it